src/views/LimitedRestorationExtension.vue

```diff
@@ -156,16 +186,7 @@ export default class Restoration extends Vue {
 
   /** Whether App is ready. */
   @Prop({ default: false }) readonly appReady!: boolean
-
-  /** The id of the restoration being edited. */
-  get restorationId (): number {
-    return +this.$route.query['restoration-id'] || 0
-  }
-
-  /** True if user is authenticated. */
-  get isAuthenticated (): boolean {
-    return Boolean(sessionStorage.getItem(SessionStorageKeys.KeyCloakToken))
-  }
+  @Prop({ required: true }) readonly restorationId!: number
```

Comment:
Passing the `restorationId` into the component as a prop

Reply:
I believe I'm casting it to a number which I thought might be more predicable downstream instead of allowing a string and having to verify the input.  I know anything on the front-end is inherently untrustworthy, but might as well make the inputs as predictable as possible.

Reply:
I'm not confident that specifying a type on the prop makes it actually that type. Or I didn't see where you cast it to a number?

Reply:
Good call, I didn't know that!  I've changed the prop from number to string then added the following check:

```javascript
    props: route => ({
      restorationId: route.query['restoration-id'].match(/\d+/)[0]
    }),
```

Reply:
Can confirm Sev's comment. I've had once to deal with that. I specified a prop as a number but it was being read as a string in one of the methods!

Reply:
I'm happy with this code now.